src/build123d/build_generic.py

```diff
@@ -94,13 +94,14 @@ def __init__(
         context.validate_inputs(self, objects)
 
         if isinstance(context, BuildPart):
-            rotation_value = (0, 0, 0) if rotation is None else rotation
-            rotate = (
-                Rotation(*rotation_value)
-                if isinstance(rotation_value, tuple)
-                else rotation
-            )
-            objects = [obj.moved(rotate) for obj in objects]
+            if rotation is None:
+                rotation = Rotation(0, 0, 0)
+            elif isinstance(rotation, float):
+                raise NotImplementedError()
```

Comment:
please correct me if i am wrong, but it looks like the the float option wasn't handled in the prior version too.
obj.moved takes a Location object and uses Location.wrapped which would not exist on floats. What should the desired behavior be here?

Reply:
The float option only applies to the BuildSketch section below. In general there isn't much input validation done in any of the APIs but there probably should be a check here, something like:
```py
            if rotation is None:
                rotation = Rotation(0, 0, 0)
            elif isinstance(rotation, float):
                raise ValueError("Float values of rotation are not valid for BuildPart")

            rotate = (
                Rotation(*rotation)
                if isinstance(rotation_value, tuple)
                else rotation
            )
            objects = [obj.moved(rotate) for obj in objects]
```